scripts/eosio_build_fedora.sh

```diff
@@ -1,500 +1,252 @@
-		then
-			printf "\\n\\tUnable to create directory %s/llvm-compiler/llvm/build.\\n" "${TEMP_DIR}"
-			printf "\\n\\tExiting now.\\n"
-			exit 1;
-		fi
-		if ! cd "${TEMP_DIR}/llvm-compiler/llvm/build"
-		then
-			printf "\\n\\tUnable to enter directory %s/llvm-compiler/llvm/build.\\n" "${TEMP_DIR}"
-			printf "\\n\\tExiting now.\\n"
-			exit 1;
-		fi
-		if ! cmake -G "Unix Makefiles" -DCMAKE_INSTALL_PREFIX="${HOME}/opt/wasm" -DLLVM_ENABLE_RTTI=1 \
-		-DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly -DCMAKE_BUILD_TYPE=Release ../
-		then
-			printf "\\tCmake compiling LLVM/Clang with WASM support has exited with the above errors.\\n"
-			printf "\\n\\tExiting now.\\n"
-			exit 1;
-		fi
-		if ! make -j"${JOBS}"
-		then
-			printf "\\tMake compiling LLVM/Clang with WASM support has exited with the above errors.\\n"
-			printf "\\n\\tExiting now.\\n"
-			exit 1;
-		fi
-		if ! make install
-		then
-			printf "\\tMake installing LLVM/Clang with WASM support has exited with the above errors.\\n"
-			printf "\\n\\tExiting now.\\n"
-			exit 1;
-		fi
-		if ! rm -rf "${TEMP_DIR}/llvm-compiler" 2>/dev/null
-		then
-			printf "\\n\\tUnable to remove directory %s/llvm-compiler.\\n" "${TEMP_DIR}"
-			printf "\\n\\tExiting now.\\n"
-			exit 1;
-		fi
-		printf "\\n\\tWASM successfully installed at %s/opt/wasm\\n\\n" "${HOME}"
+		break;;
+		[Nn]* ) echo "Proceeding without update!";;
+		* ) echo "Please type 1 for yes or 2 for no.";;
+	esac
+done
+
+printf "\\n"
+
+printf "Checking RPM for installed dependencies...\\n"
+for (( i=0; i<${#DEP_ARRAY[@]}; i++ )); do
+	pkg=$( rpm -qi "${DEP_ARRAY[$i]}" 2>/dev/null | grep Name )
+	if [[ -z $pkg ]]; then
+		DEP=$DEP" ${DEP_ARRAY[$i]} "
+		DISPLAY="${DISPLAY}${COUNT}. ${DEP_ARRAY[$i]}\\n"
+		printf "!! Package %s ${bldred} NOT ${txtrst} found !!\\n" "${DEP_ARRAY[$i]}"
+		(( COUNT++ ))
 	else
-		printf "\\n\\tWASM found @ %s/opt/wasm\\n\\n" "${HOME}"
+		printf " - Package %s found.\\n" "${DEP_ARRAY[$i]}"
+		continue
 	fi
-
-	function print_instructions()
-	{
-		printf "\\n\\t%s -f %s &\\n" "$( command -v mongod )" "${MONGOD_CONF}"
-		printf "\\tcd %s; make test\\n\\n" "${BUILD_DIR}"
+done
+if [ "${COUNT}" -gt 1 ]; then
+	printf "The following dependencies are required to install EOSIO.\\n"
+	printf "${DISPLAY}\\n"
+	printf "Do you wish to install these dependencies?\\n"
+	select yn in "Yes" "No"; do
+		case $yn in
+			[Yy]* )
+				printf "Installing dependencies\\n\\n"
+				if ! sudo "${YUM}" -y install ${DEP}; then
+					printf "!! YUM dependency installation failed !!\\n"
+					printf "Exiting now.\\n"
+					exit 1;
+				else
+					printf "YUM dependencies installed successfully.\\n"
+				fi
+			break;;
+			[Nn]* ) echo "User aborting installation of required dependencies, Exiting now."; exit;;
+			* ) echo "Please type 1 for yes or 2 for no.";;
+		esac
+	done
+else
+	printf " - No required YUM dependencies to install.\\n"
+fi
+
+
+printf "\\n"
+
+
+printf "Checking CMAKE installation...\\n"
```

Comment:
Is there a reason we are building cmake on platforms we haven't needed to before?

Reply:
Versions of cmake for fedora differ greatly. Consistency and copy and paste ability between build files is why I chose to do this; we shouldn't need to worry about keep track of differences. Also, I had a lot of problems which are fixed by using the same cmake version we use for all other distros.

Reply:
I commented on the fedora one but we're also now building cmake on ubuntu & macos -- both platforms we did not do this in the past either.

As it is we only support 3.5 and newer. Can you elaborate on what the issues are? We know 3.5, 3.10, 3.12, and 3.13 work today on the develop branch. I do not think we should be supplanting software that the OS or brew provides unless we need to.

Reply:
Unfortunately the cmake which brew provides is not guaranteed to stay at 3.13 (which is the current version). There is also no cmake@3.10 or anything like that. I've had the brew folks upgrade a version of something we were dependent on at my last job and refuse to create the @ older version formula or accept the one I submitted for them. I'm looking to avoid our reliance on brew as much as possible. Not only does this make it easier to maintain long term, but we can be assured that things won't randomly stop working due to a homebrew formula upgrade we aren't notified of.